### PR TITLE
Add actions cache example.

### PR DESCRIPTION
**Pull request by welteki**
Signed-off-by: Han Verstraete (OpenFaaS Ltd) <han@openfaas.com>

## Description

Add an example on how to use the GitHub Actions cache with actuated.

## Motivation and Context

Show users it is possible to use the Actions cache with Actuated runners.

## How Has This Been Tested?

Verified the docs locally. Workflows where tested when writing the associated blog post.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

File: docs/examples/github-actions-cache.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Jobs on Actuated runners start in a clean VM each time. This means dependencies need to be downloaded and build artifacts or caches rebuilt each time. Caching these files in the actions cache can improve workflow execution time.
+
+A lot of the setup actions for package managers have support for caching built-in. See: [setup-node](https://github.com/actions/setup-node#caching-global-packages-data), [setup-python](https://github.com/actions/setup-python#caching-packages-dependencies),... . They require minimal configuration and will create and restore dependency caches for you.
+
+If you have custom workflows that could benefit from caching the cache can be configured manually using the [actions/cache](https://github.com/actions/cache).
+
+Using the actions cache is not limited to GitHub hosted runners but can be used with self-hosted runners. Workflows using the cache action can be converted to run on Actuated runners. You only need to change `runs-on: ubuntu-latest` to `runs-on: actuated`.
+
+## Try out the action on your agent
```
Comment (alexellis) [inline docs/examples/github-actions-cache.md:11]: Enable caching on a supported action.

File: docs/examples/github-actions-cache.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          go-version: ~1.19
+          cache: true
+      - name: Build
+        run: |
+          CGO_ENABLED=0 GO111MODULE=on \
+          go build -ldflags "-s -w -X main.Release=dev -X main.SHA=dev" -o controller
+```
+
+To configure caching with the [setup-go action](https://github.com/actions/setup-go) you only need to set the `cache` input parameter to true.
+
```
Comment (alexellis) [inline docs/examples/github-actions-cache.md:41]: Add another header to show how to cache without the built-in support. Like we did in the blog post.

File: docs/examples/github-actions-cache.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Example: GitHub Actions cache
+
+Jobs on Actuated runners start in a clean VM each time. This means dependencies need to be downloaded and build artifacts or caches rebuilt each time. Caching these files in the actions cache can improve workflow execution time.
+
+A lot of the setup actions for package managers have support for caching built-in. See: [setup-node](https://github.com/actions/setup-node#caching-global-packages-data), [setup-python](https://github.com/actions/setup-python#caching-packages-dependencies),... . They require minimal configuration and will create and restore dependency caches for you.
```
Comment (alexellis) [inline docs/examples/github-actions-cache.md:5]: Is there support for setup-rust? @Addyvan was asking about that

Comment (welteki) [reply docs/examples/github-actions-cache.md:5]: There is no official setup-rust action. Not sure if the actions from [actions-rs](https://github.com/actions-rs) support it.

There is [an example](https://github.com/actions/cache/blob/main/examples.md#rust---cargo) to configure caching with rust available in the [actions/cache](https://github.com/actions/cache) repo.

**Comment by welteki** [comment]
Included an additional example for manual configuration and added a link to the configuration examples for different languages and frameworks.

**Review by alexellis — APPROVED** [review]
LGTM

File: docs/examples/github-actions-cache.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Build
+        run: |
+          CGO_ENABLED=0 GO111MODULE=on \
+          go build -ldflags "-s -w -X main.Release=dev -X main.SHA=dev" -o controller
+```
+
+The setup `Setup Golang caches` uses the [cache action](https://github.com/actions/cache) to configure caching.
```
Comment (alexellis) [inline docs/examples/github-actions-cache.md:85]: I think we could explain how you built up the `key` field better?

**Pull request merged**